src/lib/loaders/index.ts

```diff
@@ -61,25 +61,35 @@ export interface LoaderFactory {
   ): DynamicPathLoader<BodyAndHeaders<Body, Headers>, PathGeneratorParams>
 }
 
+type loadersAtRoot = LoadersWithoutAuthentication &
+  Partial<LoadersWithAuthentication>
+
+interface availableLoaders extends loadersAtRoot {
+  unauthenticatedLoaders: LoadersWithoutAuthentication
+  authenticatedLoaders: Partial<LoadersWithAuthentication>
```

Comment:
This property should be optional and the loaders should, unlike those that are spread, not be Partial.

Reply:
I don't think it should be optional, since having a `authenticatedLoaders: {}` key is very important to be able to destructure it. If the key was completely missing it would be messer in the resolver.

Reply:
`{ authenticatedLoaders: { someLoader, ... } }` would cause crashes, you'd always need to reference `context.authenticatedLoaders && ` and stuff like that...it would really muddy up the code IMO.

That's why I left it as `Partial`, and included a `let authenticatedLoaders = {}` in the block.

What do you think?

Reply:
Ahh I see, nice one. Yeah then this is correct 👍